src/tiled/mainwindow.cpp

```diff
@@ -1129,6 +1135,46 @@ void MainWindow::setFullScreen(bool fullScreen)
         setWindowState(windowState() & ~Qt::WindowFullScreen);
 }
 
+void MainWindow::toggleClearView(bool clearView)
+{
+    //Static vectors keep list of visible docks and toolbars before toggle
+    static std::vector<bool> visibleDocks, visibleToolbars;
+    QList<QDockWidget*> docks = this->findChildren<QDockWidget*>();
+    QList<QToolBar*> toolbars = this->findChildren<QToolBar*>();
+
+    if(clearView)
+    {
+        visibleDocks.clear();
+        visibleToolbars.clear();
+        mDocumentManager->saveState();
+        for(auto dock : docks)
+        {
+            visibleDocks.push_back(dock->isVisible());
+            dock->hide();
+        }
+        for(auto toolbar : toolbars)
+        {
+            visibleToolbars.push_back(toolbar->isVisible());
+            toolbar->setVisible(false);
+        }
+    }
+    else
+    {
+        if(visibleDocks.empty() || visibleToolbars.empty())
+            return;
```

Comment:
That was originally to prevent from `visibleDocks` and `visibleToolbars` throwing an exception if uninitialized during loop lookup by index, but with appropriate `QList`-s that would be deprecated